x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkStep.java

```diff
@@ -98,10 +98,37 @@ public void performAction(
         // Hard coding this to true as the resize request was executed and the corresponding cluster change was committed, so the
         // eventual retry will not be able to succeed anymore (shrunk index was created already)
         // The next step in the ShrinkAction will wait for the shrunk index to be created and for the shards to be allocated.
-        getClient().admin().indices().resizeIndex(resizeRequest, listener.delegateFailureAndWrap((l, response) -> l.onResponse(null)));
-
+        getClient().admin().indices().resizeIndex(resizeRequest, listener.delegateFailure((l, response) -> {
+            ClusterStateObserver.Listener clusterStateListener = new ClusterStateObserver.Listener() {
+                @Override
+                public void onNewClusterState(ClusterState state) {
```

Comment:
This is not the right way to use an observer, instead you should pass the condition as a predicate to `waitForNextChange`.